### PR TITLE
Add Chroma datastore support

### LINKED ISSUE #60: Support for Chroma vector datastore
[Chroma](https://trychroma.com) is an open-source embedding database and AI tooling library. Support it as one of the various vector datastore options we provide.

### PR DESCRIPTION
**Pull request by csvoss**
This adds vector datastore support for [Chroma](https://trychroma.com), an open-source embedding database. Resolves #60.

cc @atroyn pls review!

---

* [x] Test this code end-to-end
  * [x] Make sure it can run with `poetry run start`!
  * [x] Write some integration tests
* [x] Resolve uncertainties (see `TODO(csvoss)` in the current PR) about which embeddings function to use. The outer DataStore wrapper provides its own scaffolding for invoking embeddings on each query, because it generates `QueryWithEmbedding` objects that then get passed to our child class's `_query` method. However, the Chroma client also accepts an `embedding_function`.
  * [x] Pick one of these embedding functions that should be the source-of-truth method that the Chroma datastore implementation handles embeddings
  * [x] Make sure that embeddings are _only_ handled by that source-of-truth method, and that the code paths do not result in embeddings being created more than once by both methods (which would be more expensive!)

### CODE REVIEW

File: datastore/providers/chroma_datastore.py
Comment (isafulf) [inline datastore/providers/chroma_datastore.py:167]: Could you comment this out for now? Returning the whole embedding makes the response too large for ChatGPT (results in ResponseTooLarge error because of size limits on the plugin responses)

I should make this clearer in the documentation, and also plan to add an include_embeddings option to the query endpoint for all the vector dbs because it seems like some people are using this as a general retrieval API and not just as a ChatGPT plugin

**Review by isafulf — APPROVED** [review]
Thank you for adding this Chelsea!! I'll wait for Anton's changes before merging

**Comment by Bardia95** [comment]
@csvoss lmk when this is shipping, trying to use chroma for a chatgpt plugin template to publish on replit

**Comment by atroyn** [comment]
@Bardia95 this is currently blocked on me, we're refactoring some internal chroma stuff before going ahead 
@csvoss, we might want to move this into draft until that lands, expected in the next couple of days

**Comment by csvoss** [comment]
Freshly rebased off `main`. Marking as draft per the above!

File: datastore/providers/chroma_datastore.py
Comment (icaptadel) [reply datastore/providers/chroma_datastore.py:167]: o can you help

**Review by icaptadel — COMMENTED** [review]
ad

**Comment by sdan** [comment]
+1 any reason this hasn't been merged?

**Comment by atroyn** [comment]
This should be deprecated in favor of https://github.com/openai/chatgpt-retrieval-plugin/pull/232

**Comment by csvoss** [comment]
Deprecated in favor of https://github.com/openai/chatgpt-retrieval-plugin/pull/232 !

**Pull request closed without merging**
